external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriter.scala

```diff
@@ -50,7 +50,7 @@ private[kafka010] object KafkaWriter extends Logging {
       topic: Option[String] = None): Unit = {
     schema.find(_.name == TOPIC_ATTRIBUTE_NAME).getOrElse(
       if (topic.isEmpty) {
-        throw new AnalysisException(s"topic option required when no " +
+        throw new IllegalArgumentException(s"topic option required when no " +
```

Comment:
Now we check the options at the physical plan phase, so this should not be `AnalysisException` anymore.

Reply:
If this is unacceptable, then we may need to have analysis time write info and runtime write info. `Table.newWriteBuilder` takes analysis time write info and `WriteBuilder.build` takes runtime write info.

I'm not sure if it's worth this complexity.

Reply:
Should it be SparkException? I think the last time we discussed these, it wasn't clear what type of exception to use after analysis. Maybe we need new exception types?

Reply:
We can use `SparkException` as well. `IllegalArgumentException` is a standard java exception which indicates invalid input, I think it's OK to use it even after analysis.

Reply:
I think we typically want to always raise SparkException because all exception types inherit from it. Unless we are throwing an exception from a method where there is an illegal argument, but that's not what is happening here.

Reply:
> I think we typically want to always raise SparkException because all exception types inherit from it.

In Spark SQL no exceptions inherit from it. In fact SparkException was rarely used in Spark SQL before we adding the v2 commands. `SparkException` is defined in spark core and usually used when Spark fails to run a task.

In Spark SQL, AnalysisException and standard Java exceptions are more widely used.

Reply:
Sorry, I thought that AnalysisException inherited from SparkException. Looks like I was wrong.